test/e2e/specs/editor/various/toolbar-roving-tabindex.spec.js

```diff
@@ -0,0 +1,216 @@
+			'Body cell text',
+			'Table'
+		);
+		await ToolbarRovingTabindexUtils.wrapCurrentBlockWithGroup( 'Table' );
+		await ToolbarRovingTabindexUtils.testGroupKeyboardNavigation(
+			'Block: Table',
+			'Table'
+		);
+	} );
+
+	test( 'ensures custom html block toolbar uses roving tabindex', async ( {
+		editor,
+		ToolbarRovingTabindexUtils,
+	} ) => {
+		await editor.insertBlock( { name: 'core/html' } );
+		await ToolbarRovingTabindexUtils.testBlockToolbarKeyboardNavigation(
+			'HTML',
+			'Custom HTML'
+		);
+		await ToolbarRovingTabindexUtils.wrapCurrentBlockWithGroup(
+			'Custom HTML'
+		);
+		await ToolbarRovingTabindexUtils.testGroupKeyboardNavigation(
+			'Block: Custom HTML',
+			'Custom HTML'
+		);
+	} );
+
+	test( 'ensures block toolbar remembers the last focused item', async ( {
+		editor,
+		page,
+		pageUtils,
+		ToolbarRovingTabindexUtils,
+	} ) => {
+		await editor.insertBlock( { name: 'core/paragraph' } );
+		await page.keyboard.type( 'Paragraph' );
+		await ToolbarRovingTabindexUtils.focusBlockToolbar();
+		await page.keyboard.press( 'ArrowRight' );
+		await ToolbarRovingTabindexUtils.expectLabelToHaveFocus( 'Move up' );
+		await page.keyboard.press( 'Tab' );
+		await pageUtils.pressKeyWithModifier( 'shift', 'Tab' );
+		await ToolbarRovingTabindexUtils.expectLabelToHaveFocus( 'Move up' );
+	} );
+
+	test( 'can reach toolbar items with arrow keys after pressing alt+F10', async ( {
+		page,
+		pageUtils,
+		ToolbarRovingTabindexUtils,
+	} ) => {
+		await pageUtils.pressKeyWithModifier( 'alt', 'F10' );
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.press( 'ArrowRight' );
+		await ToolbarRovingTabindexUtils.expectLabelToHaveFocus( 'Bold' );
+	} );
+} );
+
+class ToolbarRovingTabindexUtils {
+	constructor( { page, pageUtils } ) {
+		this.page = page;
+		this.pageUtils = pageUtils;
+	}
+
+	async focusBlockToolbar() {
+		await this.pageUtils.pressKeyWithModifier( 'alt', 'F10' );
+	}
+
+	async testBlockToolbarKeyboardNavigation(
+		currentBlockLabel,
+		currentBlockTitle
+	) {
+		await this.focusBlockToolbar();
+		await this.expectLabelToHaveFocus( currentBlockTitle );
+		await this.page.keyboard.press( 'ArrowRight' );
+		await this.expectLabelToHaveFocus( 'Move up' );
+		await this.page.keyboard.press( 'Tab' );
+		await this.expectLabelToHaveFocus( currentBlockLabel );
+		await this.pageUtils.pressKeyWithModifier( 'shift', 'Tab' );
+		await this.expectLabelToHaveFocus( 'Move up' );
+	}
+
+	async expectLabelToHaveFocus( label ) {
+		let ariaLabel = await this.page.evaluate( () =>
+			document.activeElement.getAttribute( 'aria-label' )
+		);
+		// If the labels don't match, try pressing Up Arrow to focus the block wrapper in non-content editable block.
+		if ( ariaLabel !== label ) {
+			await this.page.keyboard.press( 'ArrowUp' );
+			ariaLabel = await this.page.evaluate( () =>
+				document.activeElement.getAttribute( 'aria-label' )
+			);
+		}
```

Comment:
I couldn't update this section to an accessible implementation. This implementation also exists in [the testing of image block](https://github.com/WordPress/gutenberg/blob/4b4c4befb34d815634b85cbee23cad169ab0e073/test/e2e/specs/editor/blocks/image.spec.js#L222-L226).